assets/src/components/admin/admin_add_modal.tsx

```diff
@@ -90,8 +90,7 @@ const AddModal = ({ setData, closeModal }): JSX.Element => {
   const addScreen = () => {
     const newRow = {
       app_id: formValues.app_id,
-      // @ts-expect-error
-      app_params: defaultAppParamsByAppId[formValues.app_id],
+      app_params: defaultAppParamsByAppId[formValues.app_id ?? 0],
```

Comment:
I'm confused as to what this would do in the error case, or why the compiler is fine with this — there is no key `0` in the object.

Reply:
This was a bad way to handle this in hindsight. I changed it a bit so `useState` is called with the type of `initialFormValues`. And since each field in `fields` is a string, I changed the initial values from `undefined` to `string`.